provisioning/transport/mqtt/src/ProvisioningChannelHandlerAdapter.cs

```diff
@@ -344,11 +374,12 @@ private async Task ProcessRegistrationStatusAsync(IChannelHandlerContext context
                 //"{\"operationId\":\"0.indcertdevice1.e50c0fa7-8b9b-4b3d-8374-02d71377886f\",\"status\":\"assigning\"}"
                 var operation = JsonConvert.DeserializeObject<RegistrationOperationStatus>(jsonData);
                 operationId = operation.OperationId;
+                operation.RetryAfter = getRetryAfterFromTopic(packet.TopicName);
 
                 if (string.CompareOrdinal(operation.Status, RegistrationOperationStatus.OperationStatusAssigning) == 0 ||
                     string.CompareOrdinal(operation.Status, RegistrationOperationStatus.OperationStatusUnassigned) == 0)
                 {
-                    await Task.Delay(s_defaultOperationPoolingIntervalMilliseconds).ConfigureAwait(true);
+                    await Task.Delay(operation.RetryAfter ?? s_defaultOperationPoolingIntervalMilliseconds).ConfigureAwait(true);
```

Comment:
This is dangerous:
1. Please add a check to ensure a minimum retry time. That will avoid high-CPU spinning if the RetryAfter is too short due to bugs.
2. Check if RetryAfter is negative.

For violations of above rules we need to log a Warn message.

Reply:
Also, we should add some jitter if RetryAfter is not present. (We do see this as a bug in the minimum time implementation in DeviceClient.)

Reply:
Why would we need to validate a value provided by the service here? If the service does not provide a retryafter value, we assume a 2 second delay. The service should never provide a value to causes high-CPU spinning

Reply:
Sure we can validate the value provided by service, and add some jitter to the default retry